packages/studio-be/src/studio/cloud/cloud-router.ts

```diff
@@ -25,19 +50,103 @@ export class CloudRouter extends CustomStudioRouter {
           scope: 'nlu'
         })
       )
-      .then((res) => `Bearer ${res.data?.access_token}`)
-      .catch((err) => null)
+
+      if (!data.access_token) {
```

Comment:
So if I understand correctly, it's possible that the call to the OAuth endpoint returns a 2XX code without giving an access token?

Reply:
that was Already there I didn't change the logic